fixtures/dev-env/tests/index.test.ts

```diff
@@ -649,3 +649,42 @@ describe("startDevWorker: ProxyController", () => {
 		await expect(res.text()).resolves.toBe("UserWorker:3");
 	});
 });
+
+describe("regressions", () => {
+	// https://github.com/cloudflare/workers-sdk/issues/5095
+	test("#5095 requests where body (of >= 2 bytes) is not drained sometimes fail", async () => {
+		const run = await fakeStartUserWorker({
+			script: `
+				export default {
+					async fetch(request) {
+						return new Response("UserWorker:1");
+					}
+				}
+			`,
+		});
+
+		const COUNT = 30;
+		const STEP_DELAY = 4; // setting to less than 4 results in errorCount === 0 indicating race condition or a request influencing a subsequent request
+		let requestCount = 0;
+		let errorCount = 0;
+
+		await Promise.allSettled(
+			Array.from({ length: COUNT }).map(async (_, i) => {
+				await new Promise((r) => setTimeout(r, i * STEP_DELAY));
+
+				return undici
+					.fetch(run.proxyWorkerUrl, {
+						method: "POST",
+						body: new Uint8Array(2), // 2 bytes
+					})
+					.then((res) => {
+						if (!res.ok) errorCount++;
+						requestCount++;
+					});
+			})
+		);
```

Comment:
Is there a reason this can't be a for loop?

Reply:
Yeah this can be simpler and still trigger the regression:

```ts
		for (let i = 0; i < COUNT; i++) {
			const response = await undici.fetch(run.proxyWorkerUrl, {
				method: "POST",
				body: new Uint8Array(2), // 2 bytes
			});
			requestCount++;
			if (!response.ok) errorCount++;
		}
```

Reply:
This is what I wrote originally. But I wanted to see if parallel requests would cause the problem to occur more often or less often. My hypothesis was the former.

In fact, if you send them within quick succession (all at once, or less than 4ms between each), the problem doesn't even manifest.

In conclusion, I don't think the problem is as simple as "request bodies need to be drained". The workaround in this PR helps. But either it delays a race condition or it releases pressure on reaching a memory limit.